src/main/java/paymelah/logic/parser/DeleteDebtCommandParser.java

```diff
@@ -0,0 +1,49 @@
+package paymelah.logic.parser;
+
+import static paymelah.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static paymelah.logic.parser.CliSyntax.PREFIX_DEBT;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import paymelah.commons.core.index.Index;
+import paymelah.logic.commands.DeleteDebtCommand;
+import paymelah.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteDebtCommand object
+ */
+public class DeleteDebtCommandParser implements Parser<DeleteDebtCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteDebtCommand
+     * and returns a DeleteDebtCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteDebtCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_DEBT);
+
+        if (argMultimap.getPreamble().isEmpty()
+                || !arePrefixesPresent(argMultimap, PREFIX_DEBT)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteDebtCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            Index personIndex = ParserUtil.parseIndex(argMultimap.getPreamble());
+            Set<Index> indexList = ParserUtil.parseIndexes(argMultimap.getAllValues(PREFIX_DEBT));
+            return new DeleteDebtCommand(personIndex, indexList);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteDebtCommand.MESSAGE_USAGE), pe);
+        }
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
@WR3nd3 did u forget to pull and merge the master before ur initial PR commit :')
but yes, I hv previously relocated this method into ParserUtil as a public static method since multiple Commands use it 
since it doesn't make sense to have it defined in several classes.